spec/services/idv/usps_mail_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'rails_helper'
+
+describe Idv::UspsMail do
+  describe '#mail_spammed?' do
+    context 'when no mail has been sent' do
+      it 'is never spammed' do
+        user = create(:user)
+        subject = Idv::UspsMail.new(user)
+        expect(subject.mail_spammed?).to eq false
+      end
+    end
+
+    context 'when the max pieces of mail have been sent' do
+      it 'is spammed if all the updates have been within the last month' do
+        user = create(:user)
+        subject = Idv::UspsMail.new(user)
+
+        events = Array.new(4) do
+          Event.create(event_type: :usps_mail_sent, user_id: user.id)
+        end
+
+        allow(subject).to receive(:find_user_events).with(:usps_mail_sent, user).
+          and_return(events.reverse)
+        expect(subject.mail_spammed?).to eq true
+      end
+
+      it 'is not spammed if the most distant update was more than a month ago' do
+        user = create(:user)
+        subject = Idv::UspsMail.new(user)
+
+        events = Array.new(4) do
+          Event.create(event_type: :usps_mail_sent, user_id: user.id)
+        end
+        events = events.reverse
+        events.last.updated_at = 2.months.ago
+        events.last.created_at = 2.months.ago
+
+        allow(subject).to receive(:find_user_events).with(:usps_mail_sent, user).
```

Comment:
Ok so the tests in this file should not be stubbing like this if we're actually creating stuff in the database.

It looks like we're creating events above and modifying the timestamps but not saving them to the DB. What I would do is change the vent creation to be:

```ruby
Event.create(event_type: :usps_mail_sent, user: user, created_at: 2.months.ago)
Event.create(event_type: :usps_mail_sent, user: user, created_at: 1.week.ago)
Event.create(event_type: :usps_mail_sent, user: user, created_at: 1.day.ago)
Event.create(event_type: :usps_mail_sent, user: user, created_at: 1.hour.ago)
```

and remove the `allow(subject).to receive` entirely